ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/fetch/fetch.tsx

```diff
@@ -10,13 +10,33 @@
  *
  **/
 
-type Param2 = {
+type Options = {
   headers?: object
   [key: string]: unknown
 }
 
-export default function(url: string, { headers, ...opts }: Param2 = {}) {
-  return window.fetch(url, {
+const fetch = window.fetch
+// @ts-ignore
+window.__global__fetch = fetch
+
+// patch global fetch to warn about usage during development
+if (process.env.NODE_ENV !== 'production') {
+  window.fetch = (...args) => {
+    const error = new Error(
+      [
+        `Using 'window.fetch'.`,
+        'Are you sure you want to do this?',
+        `Most code should use 'react-component/utils/fetch' which provides defaults compatible with the backend.`,
```

Comment:
What is the issue with a developer directly using window.fetch? Isn't that what they would be doing if the imported from the directory specified above?

Reply:
The wrapper appends required headers for CSRF and will eventually be the place we centralize network failure policies.

Reply:
should we just override window.fetch or are we concerned that it will break third party libraries/etc

Reply:
I think there are cases where we're not communicating with the DDF backend itself, thus do not want to append CSRF headers, and need the default behavior of `window.fetch` for things to work correctly. I think developers need the flexibility but this check that was added helps ensure we're making the right choices, given that flexibility.

Reply:
I want people to opt-in for the case @Lambeaux made